src/Aspire.Hosting.Azure/AzureResourcePreparer.cs

```diff
@@ -376,4 +381,62 @@ private static void AppendAppliedRoleAssignmentsAnnotation(IResource resource, I
             resource.Annotations.Add(new AppliedRoleAssignmentsAnnotation([.. newRoles]));
         }
     }
+
+    private static void CreateGlobalRoleAssignments(DistributedApplicationModel appModel, AzureProvisioningOptions provisioningOptions)
+    {
+        var azureResourceSnapshot = appModel.Resources.OfType<AzureProvisioningResource>().ToArray(); // avoid modifying the collection while iterating
+
+        foreach (var azureResource in azureResourceSnapshot)
+        {
+            if (azureResource.TryGetLastAnnotation<AppliedRoleAssignmentsAnnotation>(out var appliedRoleAssignments))
```

Comment:
With this change, we can make the `AppliedRoleAssignmentsAnnotation` internal, and rename it to anything we want.

Right now the annotaiton is still needed to support the case where you have 2 apps targeting an Azure resource, and one app requires role X and the other app requires role Y. Here, we will generate a single file `{azureResource}-roles` with all the specified roles. The annotation is used to gather all the targeted roles in the first step, and then later during `CreateGlobalRoleAssignments`, use the annotations to generate the bicep.

The other option to support this scenario would be to generate 2 role files `{app1}-roles-{azureResource}` and `{app2}-roles-{azureResource}`.

Reply:
Rethinking this, I'm not sure we should make this change. I think it may be too breaking since someone could be using ConfigureInfrastructure on their Azure resource, and moving these role assignments out to a different bicep module would mean their code would no longer work.

Reply:
Does this mean that every new azure resource also needs this logic forever? When we do remove it?

Reply:
Moving forward with this change means that every new azure resource won't need this logic. The `AppliedRoleAssignmentsAnnotation` will become internal, and can eventually be removed on more refactoring.